access/types/forms.py

```diff
@@ -322,12 +322,18 @@ def grade(self):
                 prev = i
                 i, ok, p = self.grade_field(i, field)
                 points += p
-                if not ok:
+                if self.exercise["feedback"]:
+                    self.fields[self.field_name(prev, field)].grade_points = field.get('points', 0)
```

Comment:
I think that `self.fields[...].grade_points` should be set in the `grade_field` method since it is normally done there. Why should it be done here for feedback exercises separately?

Reply:
Indeed if the ok value was changed in the grade_field, this if else statement could be more clear and it would do the grading where it should be done.